rest.go

```diff
@@ -0,0 +1,148 @@
+// Package rest allows for quick and easy access any REST or REST-like API.
+package rest
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+// Version represents the current version of the rest library
+const Version = "0.1"
+
+// Method contains the supported HTTP verbs.
+type Method string
+
+// Supported HTTP verbs.
+const (
+	Get    Method = "GET"
+	Post   Method = "POST"
+	Put    Method = "PUT"
+	Patch  Method = "PATCH"
+	Delete Method = "DELETE"
+)
+
+// Request holds the request to an API Call.
+type Request struct {
+	Method      Method
+	BaseURL     string // e.g. https://api.mayadata.com
+	Headers     map[string]string
+	QueryParams map[string]string
+	Body        []byte
+}
+
+// RestError is a struct for an error handling.
+type RestError struct {
+	Response *Response
+}
+
+// Error is the implementation of the error interface.
+func (e *RestError) Error() string {
+	return e.Response.Body
```

Comment:
Can we send better error message?
Can we add status code & http headers as well?